libs/hdf-converters/src/checklist-mapper.ts

```diff
@@ -0,0 +1,454 @@
+        const parent_profile: ExecJSON.Profile = {
+          name: parentProfileName,
+          version: '1',
+          supports: [],
+          attributes: [],
+          groups: [],
+          depends: [],
+          controls: [],
+          sha256: ''
+        };
+        for (const i in original.profiles) {
+          parent_profile.depends?.push({name: original.profiles[i].name});
+          parent_profile.controls.push(...original.profiles[i].controls);
+          original.profiles[i].parent_profile = parentProfileName;
+          original.profiles[i].sha256 = generateHash(
+            JSON.stringify(original.profiles[i])
+          );
+        }
+        parent_profile.sha256 = generateHash(JSON.stringify(parent_profile));
+        original.profiles.push(parent_profile);
+        return original;
+    }
+  }
+}
+
+/**
+ * Checklist mapper
+ */
+export class ChecklistMapper extends BaseConverter {
+  mappings: MappedTransform<
+    ExecJSON.Execution & {passthrough: unknown},
+    ILookupPath
+  > = {
+    platform: {
+      name: 'Heimdall Tools',
+      release: HeimdallToolsVersion
+    },
+    version: HeimdallToolsVersion,
+    statistics: {},
+    profiles: [
+      {
+        path: 'stigs',
+        name: {path: 'header.title'},
+        version: {path: 'header.version'},
+        title: {path: 'header.title'},
+        summary: {path: 'header.description'},
+        license: {path: 'header.notice'},
+        supports: [],
+        attributes: [],
+        groups: [],
+        status: 'loaded',
+        controls: [
+          {
+            path: 'vulns',
+            key: 'id',
+            tags: {
+              default: {path: 'vulnDiscuss'},
+              gtitle: {path: 'groupTitle'},
+              rid: {path: 'ruleId'},
+              gid: {path: 'vulnNum'},
+              stigId: {path: 'ruleVersion'},
+              cci: {
+                path: 'cciRef',
+                transformer: cciRef
+              },
+              nist: {
+                path: 'cciRef',
+                transformer: nistTag
+              },
+              weight: {path: 'weight'}
+            },
+            refs: [],
+            source_location: {},
+            title: {path: 'ruleTitle'},
+            id: {path: 'vulnNum'},
+            desc: {path: 'vulnDiscuss'},
+            descriptions: [
+              {
+                data: {path: 'checkContent'},
+                label: 'check'
+              },
+              {
+                data: {path: 'fixText'},
+                label: 'fix'
+              }
+            ],
+            impact: {
+              transformer: (vulnerability: Record<string, unknown>): number =>
+                transformImpact(vulnerability)
+            },
+            code: {
+              transformer: (vulnerability: Record<string, unknown>): string =>
+                JSON.stringify(vulnerability, null, 2)
+            },
+            results: [
```

Comment:
So far, there are two main elements we could use to map 'previously hdf' data: customname and third_party_tools. The advantage of using the latter is that it is rendered in StigViewer at the bottom of the check and fix text area under the CCI-Reference marked as MISC Data. If written in a parsable method it should be easier to map back. I would lean toward XML syntax since the .ckl is XML format but would need to be mindful of readability when in StigViewer if we opt for a visable field.